File: Frontend/game-buddy-finder/src/app/models/post.ts
import { User } from './user'
import { Comment } from './comment'

export class Post {
    postId: number;                 //Server Generated
    posterUserId: number;
    content: string;
    postTime;              //Server Generated
    Poster: User;                   //Server Generated

    comments: Array<Comment>;

    public constructor(posterUserId: number, content: string) {
        this.posterUserId = posterUserId;
        this.content = content;
    }
}

File: Backend/Game Buddy Finder/Game Buddy Finder/Models/Post.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Game_Buddy_Finder.Models
{
    public class Post
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int PostId { get; set; }

        [Required]
        public int PosterUserId { get; set; }

        [Required]
        public string Content { get; set; }

        [Required]
        public DateTime PostTime { get; set; }

        [NotMapped]
        public User Poster { get; set; }
    }
}
